src/frontend/hooks/useLocation.ts

```diff
@@ -38,8 +38,68 @@ export function useLocation({
     error: undefined,
   });
 
+  const interval = React.useRef<ReturnType<typeof setInterval>>();
+  const lastLocation = React.useRef<LocationObject>();
+
   const [permissions] = useForegroundPermissions();
 
+  const debounceLocation = React.useCallback(
+    ({
+      maxDistanceInterval,
+      maxTimeInterval = 1000,
+      minTimeInterval = 200,
+    }: Omit<LocationOptions, 'minDistanceInterval'>) => {
+      return function (location: LocationObject) {
+        // The user can turn off location services via the quick settings dropdown
+        // (swiping down from the top of their phone screen) without moving away
+        // from the app. In this case the location will just stop updating and we
+        // won't know why. If we haven't had a location update for a while, we check
+        // on the provider status to see if location services are enabled, so that
+        // we can update the state with the current status
+        if (interval.current) clearInterval(interval.current);
+        interval.current = setInterval(async () => {
+          if (!(await hasServicesEnabledAsync())) {
+            lastLocation.current = undefined;
+            setLocation({
+              location,
+              error: new Error('Location service is not enabled'),
+            });
+            clearInterval(interval.current);
+            return;
+          }
+        }, LOCATION_TIMEOUT);
```

Comment:
this needs to be an interval because `distanceInterval` is being used.

If its was a timer instead of an interval the following could happen:
- Interval set to 2 m
- User does not move more than 2 meter in 10 seconds. Because of this, the location callback is not fired for 10 seconds. 
- setTimeout() would be called. location service would still be enabled, so nothing would happen...
- user turns off location service
- Since the setTimeout() has already been called, we don't check again
- this hook would then return a stale location

This would be avoided with an interval as it would continue checking until the the location Service is disables OR the component has unmounted

Reply:
We're slightly muddling two approaches here:

1. We can continually poll the location provider state with a setInterval.
2. We can listen to position updates, and if we have not had a position update in X milliseconds, then check the location provider state.

This PR kind of mixes both approaches, which to me makes it hard to reason with and potentially introduces subtle bugs.